From: Yifeng <wuyifeng@nd.com.cn>
Date: Fri, 5 Nov 2021 05:53:00 +0000
Subject: AP9-4099 Google Web Apps shows Unsupported Message

This ensures that when we open the google web apps (like docs.google.com), there is not a message at the top that says “Unsupported Browser”.
The modified content is to add 'Chrome/xx' and 'Firefox/xx' to the User-Agent.

---
 content/common/user_agent.cc | 4 ++--
 components/embedder_support/user_agent_utils.cc | 16 -----
 2 files changed, 2 insertions(+), 18 deletions(-)

diff --git a/content/common/user_agent.cc b/content/common/user_agent.cc
--- a/content/common/user_agent.cc
+++ b/content/common/user_agent.cc
@@ -325,7 +325,7 @@ std::string BuildUserAgentFromOSAndProduct(const std::string& os_info,
   std::string user_agent;
   base::StringAppendF(&user_agent,
                       "Mozilla/5.0 (%s) AppleWebKit/537.36 (KHTML, like Gecko) "
-                      "%s Safari/537.36",
-                      os_info.c_str(), "");
+                      "%s Safari/537.36 Firefox/97.2.0",
+                      os_info.c_str(), product.c_str());
   return user_agent;
 }
diff --git a/components/embedder_support/user_agent_utils.cc b/components/embedder_support/user_agent_utils.cc
--- a/components/embedder_support/user_agent_utils.cc
+++ b/components/embedder_support/user_agent_utils.cc
@@ -349,18 +349,4 @@
 std::string GetUserAgent(
     ForceMajorVersionToMinorPosition force_major_to_minor) {
-  base::CommandLine* command_line = base::CommandLine::ForCurrentProcess();
-  if (command_line->HasSwitch(kUserAgent)) {
-    std::string ua = command_line->GetSwitchValueASCII(kUserAgent);
-    if (net::HttpUtil::IsValidHeaderValue(ua))
-      return ua;
-    LOG(WARNING) << "Ignored invalid value for flag --" << kUserAgent;
-  }
-
-  if (base::FeatureList::IsEnabled(blink::features::kFullUserAgent))
-    return GetFullUserAgent();
-
-  if (base::FeatureList::IsEnabled(blink::features::kReduceUserAgent))
-    return GetReducedUserAgent(force_major_to_minor);
-
   return GetFullUserAgent(force_major_to_minor);
 }
@@ -383,10 +369,5 @@
 std::string GetFullUserAgent(
     ForceMajorVersionToMinorPosition force_major_to_minor) {
   std::string product = GetProductAndVersion(force_major_to_minor);
-#if BUILDFLAG(IS_ANDROID)
-  if (base::CommandLine::ForCurrentProcess()->HasSwitch(
-          switches::kUseMobileUserAgent))
-    product += " Mobile";
-#endif
   return content::BuildUserAgentFromProduct(product);
 }
--
2.17.1